File: platform/android/TrackasiaGLAndroidSDKTestApp/src/main/java/com/trackasia/android/testapp/utils/IdleZoomListener.java
package com.trackasia.android.testapp.utils;

import android.content.Context;
import android.widget.TextView;

import com.trackasia.android.camera.CameraPosition;
import com.trackasia.android.maps.TrackasiaMap;
import com.trackasia.android.testapp.R;

public class IdleZoomListener implements TrackasiaMap.OnCameraIdleListener {

  private TrackasiaMap mapboxMap;
  private TextView textView;

  public IdleZoomListener(TrackasiaMap mapboxMap, TextView textView) {
    this.mapboxMap = mapboxMap;
    this.textView = textView;
  }

  @Override
  public void onCameraIdle() {
    Context context = textView.getContext();
    CameraPosition position = mapboxMap.getCameraPosition();
    textView.setText(String.format(context.getString(R.string.debug_zoom), position.zoom));
  }
}

File: platform/android/TrackasiaGLAndroidSDKTestApp/src/main/java/com/trackasia/android/testapp/utils/IdleZoomListener.kt
package com.trackasia.android.testapp.utils

import android.widget.TextView
import com.trackasia.android.maps.TrackasiaMap
import com.trackasia.android.maps.TrackasiaMap.OnCameraIdleListener
import com.trackasia.android.testapp.R

class IdleZoomListener(private val trackasiaMap: TrackasiaMap, private val textView: TextView) :
    OnCameraIdleListener {
    override fun onCameraIdle() {
        val context = textView.context
        val position = trackasiaMap.cameraPosition
        textView.text = String.format(context.getString(R.string.debug_zoom), position.zoom)
    }
}
